ee/control/client_http.go

```diff
@@ -40,77 +63,150 @@ func NewControlHTTPClient(logger log.Logger, addr string, client *http.Client, o
 	return c, nil
 }
 
-func (c *HTTPClient) Get(hash string) (data io.Reader, err error) {
-	verb, path := "GET", fmt.Sprintf("/api/v1/control/%s", hash)
+func (c *HTTPClient) GetConfig() (io.Reader, error) {
+	challengeReq, err := http.NewRequest(http.MethodGet, c.url("/api/agent/config").String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not create challenge request: %w", err)
+	}
+
+	challenge, err := c.do(challengeReq)
+	if err != nil {
+		return nil, fmt.Errorf("could not make challenge request: %w", err)
+	}
 
-	response, err := c.do(verb, path)
+	configReq, err := http.NewRequest(http.MethodPost, c.url("/api/agent/config").String(), nil)
 	if err != nil {
-		level.Debug(c.logger).Log(
-			"msg", "error making request to control server endpoint",
-			"err", err,
-		)
-		return nil, err
-	}
-	defer response.Body.Close()
-
-	switch response.StatusCode {
-	case http.StatusNotFound:
-		// This could indicate an inconsistency in server data, or a client logic error
-		level.Error(c.logger).Log(
-			"msg", "got HTTP 404 making control server request",
-		)
-		return nil, err
-
-	case http.StatusNotModified:
-		// The control server sends back a 304 Not Modified status, without a body, which tells
-		// the client that the cached version of the response is still good to use
-		level.Debug(c.logger).Log(
-			"msg", "got HTTP 304 making control server request",
-		)
-		return nil, err
-	}
-
-	if response.StatusCode != http.StatusOK {
-		level.Error(c.logger).Log(
-			"msg", "got not-ok status code from control server",
-			"response_code", response.StatusCode,
-			"response_body", response.Body,
-		)
-		return nil, err
-	}
-
-	// response.Body will be closed before this function exits, get all the data now
-	body, err := io.ReadAll(response.Body)
+		return nil, fmt.Errorf("could not create config request: %w", err)
+	}
+	configReq.Header.Set(HeaderChallenge, string(challenge))
+	configReq.Header.Set("Content-Type", "application/json")
+	configReq.Header.Set("Accept", "application/json")
+
+	// Calculate first signature
+	localDbKeys := agent.LocalDbKeys()
+	if localDbKeys.Type() != "noop" {
+		key1, err := keyHeaderValue(localDbKeys)
```

Comment:
the krypto PR I have open now exposes funcs like

`PublicEcdsaToB64Der`
`SignWithTimeout`

you don't have to use them, but I'll let you know when that PR merges if interested

Reply:
Yes definitely!

Reply:
Updated to use these!